packages/clarity-js/src/data/limit.ts

```diff
@@ -0,0 +1,38 @@
+import { Check, Event, LimitData, Setting } from "@clarity-types/data";
+import * as clarity from "@src/clarity";
+import { time } from "@src/core/time";
+import * as envelope from "@src/data/envelope";
+import encode from "./encode";
+
+export let data: LimitData;
+
+export function start(): void {
+    data = { check: Check.None };
+}
+
+export function check(bytes: number): void {
+    if (data.check === Check.None) {
+        let reason = data.check;
+        reason = envelope.data.sequence >= Setting.PayloadLimit ? Check.Payload : reason;
+        reason = time() > Setting.ShutdownLimit ? Check.Shutdown : reason;
+        reason = bytes > Setting.PlaybackBytesLimit ? Check.Shutdown : reason;
+        if (reason !== data.check) {
+            trigger(reason);
+        }
+    }
+}
+
+export function trigger(reason: Check): void {
+    data.check = reason;
+    clarity.stop();
+}
```

Comment:
how do we ensure compute is called before this?

Reply:
Discussed offline. Through module ordering in data/index.ts file.